### ISSUE TITLE
Refining the definition of a multi-device user

### ISSUE THREAD

**Issue by rfk** [mozilla/fxa-activity-metrics#6]
We currently define a multi-device user on day X as someone who was active on day X, and was active with a different device-id in the previous five days.  This has been pretty successful and helps capture folks who transition from multi- to single-device and back again.

However, it does have some false positives.  If a user _replaces_ their device, there will be a few days during the transition where they meet this definition of being multi-device.  It a user logs out and logs back in on the same browser, they'll get a new device-id and will again qualify as a multi-device user for five days.

I wonder if we can eliminate these false-positives by changing the definition of multi-device to users who were:
- Active on Day X with device-id A,
- Active with device-id B sometime in the previous five days, and
- Active with device-id A sometime in the previous five days, prior to the activity with device-id B

In other words, they had a sequence of three events that went "A, B, A", so these were definitely two separate devices in use at the same time.

This will be more expensive to calculate, but perhaps not too much more.  I guess we could try it out in parallel to the existing definition and see how much of a difference it makes.

@philbooth thoughts?  (Not suggested we pursue this with any urgency, just getting it into writing while it's on my mind)

**Comment by philbooth** [comment]
> @philbooth thoughts?

Sounds good to me, can't think of any problems with it and it definitely eliminates the gotchas you mentioned.

**Comment by rfk** [comment]
Just a note that we've come some building-blocks for this now.  The daily_multi_device_users table now has `device_now` and `device_prev` columns telling us the two device ids in question.  So we should be able to find out who meets this stricter definition of multi-device-ness doing something like:

```
SELECT APPROXIMATE COUNT(DISTINCT mdu1.uid)
FROM daily_multi_device_users mdu1
INNER JOIN daily_multi_device_users mdu2
ON mdu1.uid = mdu2.uid
AND mdu1.device_now = mdu2.device_prev
AND mdu1.device_prev = mdu2.device_now
AND mdu2.day <= mdu1.day
AND mdu2.day > mdu1.day - '5 days'::INTERVAL
```

**Comment by philbooth** [comment]
This is available in `strict_daily_multi_device_users`, closing.
